plutus-ledger-api/src/PlutusLedgerApi/V3/Contexts.hs

```diff
@@ -0,0 +1,581 @@
+  deriving newtype
+    ( Haskell.Eq
+    , Haskell.Show
+    , PlutusTx.Eq
+    , PlutusTx.ToData
+    , PlutusTx.FromData
+    , PlutusTx.UnsafeFromData
+    )
+
+newtype HotCommitteeCredential = HotCommitteeCredential V2.Credential
+  deriving stock (Generic)
+  deriving (Pretty) via (PrettyShow HotCommitteeCredential)
+  deriving newtype
+    ( Haskell.Eq
+    , Haskell.Show
+    , PlutusTx.Eq
+    , PlutusTx.ToData
+    , PlutusTx.FromData
+    , PlutusTx.UnsafeFromData
+    )
+
+newtype DRepCredential = DRepCredential V2.Credential
+  deriving stock (Generic)
+  deriving (Pretty) via (PrettyShow DRepCredential)
+  deriving newtype
+    ( Haskell.Eq
+    , Haskell.Show
+    , PlutusTx.Eq
+    , PlutusTx.ToData
+    , PlutusTx.FromData
+    , PlutusTx.UnsafeFromData
+    )
+
+data DRep
+  = DRep DRepCredential
+  | DRepAlwaysAbstain
+  | DRepAlwaysNoConfidence
+  deriving stock (Generic, Haskell.Show, Haskell.Eq)
+  deriving (Pretty) via (PrettyShow DRep)
+
+instance PlutusTx.Eq DRep where
+  {-# INLINEABLE (==) #-}
+  DRep a == DRep a'                                = a PlutusTx.== a'
+  DRepAlwaysAbstain == DRepAlwaysAbstain           = Haskell.True
+  DRepAlwaysNoConfidence == DRepAlwaysNoConfidence = Haskell.True
+  _ == _                                           = Haskell.False
+
+data Delegatee
+  = DelegStake V2.PubKeyHash
+  | DelegVote DRep
+  | DelegStakeVote V2.PubKeyHash DRep
+  deriving stock (Generic, Haskell.Show, Haskell.Eq)
+  deriving (Pretty) via (PrettyShow Delegatee)
+
+instance PlutusTx.Eq Delegatee where
+  {-# INLINEABLE (==) #-}
+  DelegStake a == DelegStake a' = a PlutusTx.== a'
+  DelegVote a == DelegVote a' = a PlutusTx.== a'
+  DelegStakeVote a b == DelegStakeVote a' b' =
+    a PlutusTx.== a' PlutusTx.&& b PlutusTx.== b'
+  _ == _ = Haskell.False
+
+newtype AnchorDataHash = AnchorDataHash {getAnchorDataHash :: PlutusTx.BuiltinByteString}
+  deriving stock (Generic)
+  deriving newtype
+    ( Haskell.Eq
+    , PlutusTx.Eq
+    , PlutusTx.Ord
+    , PlutusTx.Show
+    , PlutusTx.ToData
+    , PlutusTx.FromData
+    , PlutusTx.UnsafeFromData
+    )
+  deriving
+    ( -- | from hex encoding
+      IsString
+    , -- | using hex encoding
+      Haskell.Show
+    , -- | using hex encoding
+      Pretty
+    )
+    via V2.LedgerBytes
+
+data Anchor = Anchor
+  { anchorUrl      :: PlutusTx.BuiltinByteString -- Arbitrary Url
+  , anchorDataHash :: AnchorDataHash -- Blake2b_256 of some off-chain data
+  }
+  deriving stock (Generic, Haskell.Show, Haskell.Eq)
+
+instance Pretty Anchor where
+  pretty Anchor{..} =
+    vsep
+      [ "anchorUrl:" <+> pretty anchorUrl
+      , "anchorDataHash:" <+> pretty anchorDataHash
+      ]
+
+instance PlutusTx.Eq Anchor where
+  {-# INLINEABLE (==) #-}
+  Anchor a b == Anchor a' b' = a PlutusTx.== a' PlutusTx.&& b PlutusTx.== b'
+
+data TxCert
+  = -- | Register staking credential with an optional deposit amount
+    TxCertRegStaking V2.Credential (Haskell.Maybe V2.Value)
```

Comment:
What is the rationale with qualifying Maybe, Integer, Bool , etc with `Haskell`? These datatypes should be first and foremost targetting `PlutusTx`, right? 

The fact that these `Maybe,Integer,etc` are the same for PlutusTx and Haskell is orthogonal.

I am just asking why; not that there is sth wrong with them.

Reply:
I just want to make it abundantly clear which things are from Haskell, and which things are from the Plutus Tx library. Especially since we don't always use explicit or qualified imports.